lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1126,6 +1126,36 @@ def select_datastore(self, vm_obj=None):
                         datastore_name = datastore.name
                         datastore_freespace = ds.summary.freeSpace
 
+            elif 'datastore_cluster' in self.params['disk'][0]:
+                try:
+                    podsel = vim.storageDrs.PodSelectionSpec()
+                    pod = find_obj(self.content, [vim.StoragePod], self.params['disk'][0]['datastore_cluster'])
+                except Exception:
+                    self.module.fail_json(msg="Vmware platform seems to not be compatible with datastore cluster")
+
+                if pod:
+                    podsel.storagePod = pod
+                else:
+                    self.module.fail_json(msg="Failed to find a matching cluster datastore")
+
+                storagespec = vim.storageDrs.StoragePlacementSpec()
+                storagespec.type = 'create'
+                storagespec.podSelectionSpec = podsel
+                storagespec.folder = destfolder
+                storagespec.configSpec = configspec
+                storagespec.resourcePool = self.get_resource_pool()
+                storagespec.host = self.select_host()
+
+                try:
+                    rec = self.content.storageResourceManager.RecommendDatastores(storageSpec=storagespec)
+                    rec_action = rec.recommendations[0].action[0]
+                    real_datastore_name = rec_action.destination.name
+                except Exception:
+                    self.module.fail_json(msg="Failed to get Recommend Datastores")
```

Comment:
can you send exception message to msg as it will help to debug issues in future ?